Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -855,47 +855,59 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
     
     public enum ImageFormat
     {
-        case jpeg
+        /// - parameter quality: compression quality for lossless formats (JPEG)
+        case jpeg(quality: Double)
         case png
     }
     
-    /// Saves the current chart state with the given name to the given path on
-    /// the sdcard leaving the path empty "" will put the saved file directly on
-    /// the SD card chart is saved as a PNG image, example:
-    /// saveToPath("myfilename", "foldername1/foldername2")
+    /// Saves the current chart state with the given name to the given path, examples:
+    /// save(to: "./out/filename.jpg", format: .jpeg(quality: 90.0))
+    /// save(to: "./out/filename.png", format: .png)
     ///
     /// - parameter to: path to the image to save
     /// - parameter format: the format to save
-    /// - parameter compressionQuality: compression quality for lossless formats (JPEG)
     ///
     /// - returns: `true` if the image was saved successfully
-    open func save(to path: String, format: ImageFormat, compressionQuality: Double) -> Bool
+    open func save(to path: String, format: ImageFormat) -> Bool
     {
-		guard let image = getChartImage(transparent: format != .jpeg)
-            else { return false }
+        let saveUrl = URL(fileURLWithPath: path)
+        if #available(iOS 9.0, *)
+        {
+            if saveUrl.hasDirectoryPath
+            {
+                // the path represents a directory. not include filename.
+                return false
+            }
+        }
         
-        var imageData: Data!
+        var imageData: Data?
         switch (format)
         {
         case .png:
+            guard let image = getChartImage(transparent: true) else { return false }
```

Comment:
old way, `.jpeg != format` is not available because of associated value enum. Those enum does not have `Equatable` (We have to define by myself).